test/services/hyperlink-utility.test.js

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable import/no-extraneous-dependencies */
+const { expect, test } = require('@twilio/cli-test');
+const { Command } = require('@oclif/command');
+const flush = require('flush-cache');
+
+const ORIG_ENV = { ...process.env };
+const { supportsHyperlink, convertToHyperlink } = require('../../src/services/hyperlink-utility');
+const TwilioHelp = require('../../src/services/twilio-help/custom-help');
+
+class TestCommand extends Command {
+  constructor(argv, config) {
+    super(argv, config);
+    this.id = 'api:core';
+    this.description = 'This is a dummy description';
+    this.args = [{ name: 'arg_1', description: 'argument description' }];
+  }
+}
+const testThisConfig = ({ platform, env, argv, stream }) => {
+  platform = platform || 'darwin';
+  env = env || {};
+  argv = argv || [];
+  // back up the original env
+  const oldPlatform = process.platform;
+  const oldEnv = process.env;
+  const oldArgv = process.argv;
+
+  // Inject new env properties from args
+  Object.defineProperties(process, {
+    platform: { value: platform },
+    env: { value: env },
+    argv: { value: [process.argv[0], ...argv] },
+  });
+
+  const result = supportsHyperlink(stream);
+  // restore the original env
+  Object.defineProperties(process, {
+    platform: { value: oldPlatform },
+    env: { value: oldEnv },
+    argv: { value: oldArgv },
+  });
+  return result;
+};
+
+const testLink = test
+  .loadConfig()
+  .add('help', (ctx) => new TwilioHelp(ctx.config))
+  .register('cmdTestLink', (args) => {
+    return {
+      async run(ctx) {
+        const dummyHelpCommand = new TestCommand(args, ctx.config);
+        dummyHelpCommand.docLink = 'https://twilio.com/docs/dummyCmd';
+        const help = ctx.help.formatCommand(dummyHelpCommand);
+        ctx.cmdTestLink = help
+          .split('\n')
+          .map((s) => s.trimRight())
+          .join('\n');
+      },
+    };
+  });
+
+describe('supportsHyperlink', () => {
+  describe('test hyperlink generation', () => {
+    describe('test for Mac terminals', () => {
+      afterEach(() => {
+        flush();
```

Comment:
Since the module supports-hyperlink [exports](https://github.com/jamestalmage/supports-hyperlinks/blob/66834a8a4ccc07f9af6364778fcaa0666b61eccb/index.js#L93) variables also, this flush is required to remove the cached copy of module exports.